llvm/lib/CodeGen/GlobalISel/IRTranslator.cpp

```diff
@@ -2435,6 +2435,18 @@ bool IRTranslator::translateKnownIntrinsic(const CallInst &CI, Intrinsic::ID ID,
     MIRBuilder.buildInstr(TargetOpcode::G_RESET_FPMODE, {}, {});
     return true;
   }
+  case Intrinsic::prefetch: {
+    Value *Addr = CI.getOperand(0);
+    ConstantInt *RW = cast<ConstantInt>(CI.getOperand(1));
+    ConstantInt *Locality = cast<ConstantInt>(CI.getOperand(2));
+    ConstantInt *CacheType = cast<ConstantInt>(CI.getOperand(3));
+
+    MIRBuilder.buildPrefetch(getOrCreateVReg(*Addr), RW->getZExtValue(),
+                             Locality->getZExtValue(),
+                             CacheType->getZExtValue());
```

Comment:
Should this also get a `MachineMemOperand`?

Reply:
SDNode has SDNPMemOperand. SDag lowering also creates memory operand, like this:
```
  S_PREFETCH_DATA killed %2:sgpr_64, 0, $sgpr_null, 0 :: (load (s8) from %ir.ptr, addrspace 4)
```